configtx/config.go

```diff
@@ -78,28 +78,41 @@ type standardConfigPolicy struct {
 // ConfigTx wraps a config transaction.
 type ConfigTx struct {
 	// original state of the config
-	original *cb.Config
+	original *OriginalConfig
 	// modified state of the config
-	updated *cb.Config
+	updated *UpdatedConfig
 }
 
 // New creates a new ConfigTx from a Config protobuf.
 // New will panic if given an empty config.
 func New(config *cb.Config) ConfigTx {
 	return ConfigTx{
-		original: config,
+		original: &OriginalConfig{Config: config},
 		// Clone the base config for processing updates
-		updated: proto.Clone(config).(*cb.Config),
+		updated: &UpdatedConfig{Config: proto.Clone(config).(*cb.Config)},
 	}
 }
 
-// OriginalConfig returns the original unedited config.
-func (c *ConfigTx) OriginalConfig() *cb.Config {
+// OriginalConfig wraps the original state of the config
+// and implements functions for retrieving values on it.
+type OriginalConfig struct {
+	*cb.Config
```

Comment:
q: By embedding `*cb.Config` all of the exported fields can be directly mutated. If a user decided to modify something directly, what would the impact be on the library? Can it get into an awkwardly inconsistent state?

I'm not saying it's bad to use the embedding but it does have some side effects.